### ISSUE TITLE
Prototype Pollution due to mkdirp minimist dependency.

### ISSUE THREAD

**Issue by sbhardwaj-mt** [grpc/grpc-node#1305]
### Problem description
npm audit fails with  grpc > node-pre-gyp > mkdirp >  minimist

### Reproduction steps
Give very precise steps you've discovered to reproduce your problem. If possible and applicable, provide us with a repository we can clone that contains a reproduction case. Also if possible and applicable, please include a Dockerfile that exhibits the problem if it's specific to a certain environment. Bug reports with no reproduction steps will be closed.

### Environment
 - OS name, version and architecture: [e.g. Linux Ubuntu 18.04 amd64] MacOSx
 - Node version [e.g. 8.10.0] v10.15.3
 - Node installation method [e.g. nvm]
 - If applicable, compiler version [e.g. clang 3.8.0-2ubuntu4] 
 - Package name and version [e.g. gRPC@1.12.0] grpc@1.23.3

### Additional context
Add any other context about the problem here. If possible, attach full logs. Do not try to omit anything for brevity, but instead include absolutely everything. Please try and set your operating system's locale to English, so that logs contain error messages in the English language.

**Comment by nicolasnoble** [comment]
Closing, as 1) there's not really anything we can do about it, node-pre-gyp is completely necessary for us, and 2) you haven't filled our issue template at all, and as specified in it, issues that aren't being properly filled in will be closed immediately.

**Comment by sbhardwaj-mt** [comment]
@nicolasnoble can you please at least provide a solution of update node-pre-gyp with the version where this doesn't have this vulnerability ?

**Comment by murgatroid99** [comment]
node-pre-gyp is only ever executed by a specific script in the `package.json` file. There's no way to actually trigger this exploit, so it doesn't actually matter that it's there.

**Comment by Naktibalda** [comment]
@nicolasnoble the interesting fact is that `npm audit` comes clean when executed in node-pre-gyp code.
I think that the real issues is that node-pre-gyp is not a regular dependency of grpc packages, but a bundled dependency and it locks down deeper dependencies to specific versions.

`npm update` can update mkdirp, but then `npm ls` shows errors about unmet dependencies:

```
$ npm update mkdirp --depth=5
npm WARN grpc-tools@1.8.1 No license field.

+ mkdirp@0.5.4
updated 3 packages in 1.017s
+ mkdirp@0.5.4
removed 1 package and updated 2 packages in 0.453s

$ npm ls minimist
package-name@0.0.10 /c/Users/Eckoh/sites/psp-base
├─┬ grpc@1.24.2
│ └─┬ node-pre-gyp@0.14.0
│   ├─┬ UNMET DEPENDENCY mkdirp@0.5.1
│   │ └── UNMET DEPENDENCY minimist@0.0.8
│   └─┬ rc@1.2.8
│     └── UNMET DEPENDENCY minimist@1.2.0
├─┬ grpc-tools@1.8.1
│ └─┬ node-pre-gyp@0.12.0
│   ├─┬ UNMET DEPENDENCY mkdirp@0.5.1
│   │ └── UNMET DEPENDENCY minimist@0.0.8
│   └─┬ rc@1.2.8
│     └── UNMET DEPENDENCY minimist@1.2.0
├─┬ grpc_tools_node_protoc_ts@2.5.10
│ └─┬ handlebars@4.5.3
│   └─┬ optimist@0.6.1
│     └── minimist@0.0.10
├─┬ mocha@7.1.1
│ └─┬ mkdirp@0.5.3
│   └── minimist@1.2.5
├─┬ nodemon@2.0.2
│ └─┬ update-notifier@2.5.0
│   └─┬ latest-version@3.1.0
│     └─┬ package-json@4.0.1
│       └─┬ registry-auth-token@3.4.0
│         └─┬ rc@1.2.8
│           └── minimist@1.2.5
└─┬ nyc@14.1.1
  └─┬ spawn-wrap@1.4.3
    └─┬ mkdirp@0.5.4
      └── minimist@1.2.5

npm ERR! missing: mkdirp@0.5.1, required by node-pre-gyp@0.14.0
npm ERR! missing: minimist@0.0.8, required by mkdirp@0.5.1
npm ERR! missing: minimist@1.2.0, required by rc@1.2.8
npm ERR! missing: mkdirp@0.5.1, required by node-pre-gyp@0.12.0
npm ERR! missing: minimist@0.0.8, required by mkdirp@0.5.1
npm ERR! missing: minimist@1.2.0, required by rc@1.2.8
```

**Comment by Naktibalda** [comment]
I checked out grpc 1.24.2, ran `npm install`, check that `npm audit` is green and published it as 1.24.3 to our private npm repository.
It fixed dependency issues for me:

├─┬ grpc@1.24.3
│ └─┬ node-pre-gyp@0.14.0
│   ├─┬ mkdirp@0.5.4
│   │ └── minimist@1.2.5
│   └─┬ rc@1.2.8
│     └── minimist@1.2.5  deduped
├─┬ grpc-tools@1.8.1
│ └─┬ node-pre-gyp@0.12.0
│   ├─┬ mkdirp@0.5.1
│   │ └── minimist@0.0.8
│   └─┬ rc@1.2.8
│     └── minimist@1.2.0

So, to get rid of current vulnerabilities, it is necessary to publish new versions of `grpc` and `grpc-tools` packages.
